otherlibs/cram/bin/run.ml

```diff
@@ -2,6 +2,40 @@ open Stdune
 open Cmdliner
 open Import.Let_syntax
 
+module Temp_dir : sig
+  type t
+
+  val create : for_script:string -> t
+
+  val open_file : t -> suffix:string -> string * out_channel
+
+  val file : t -> suffix:string -> string
+end = struct
+  type t = string
+
+  let prng = lazy (Random.State.make_self_init ())
+
+  let temp_file_name temp_dir prefix suffix =
+    let rnd = Random.State.bits (Lazy.force prng) land 0xFFFFFF in
+    Filename.concat temp_dir (Printf.sprintf "%s%06x%s" prefix rnd suffix)
+
+  let create ~for_script =
+    let dir = Filename.get_temp_dir_name () in
+    let t = temp_file_name dir ".dune.cram." (Filename.basename for_script) in
+    let path = Path.External.of_string t in
+    Path.External.mkdir_p path;
+    at_exit (fun () -> Path.rm_rf ~allow_external:true (Path.external_ path));
+    t
+
+  let file t ~suffix = Filename.temp_file ~temp_dir:t "" suffix
+
+  let open_file t ~suffix =
+    let fn, oc =
+      Filename.open_temp_file ~temp_dir:t "" suffix ~mode:[ Open_binary ]
```

Comment:
We don't need the randomness here. Given that we are creating all files under a temporary directory, we can choose whatever filenames we want.

BTW, we might also want to set the temporary directory variable to this directory (or a `tmp` subdir), just to keep things clean in case the commands inside the cram test don't properly clean up their temporary files.

Reply:
> We don't need the randomness here. Given that we are creating all files under a temporary directory, we can choose whatever filenames we want.

Good point. I'll remove that.

> BTW, we might also want to set the temporary directory variable to this directory (or a tmp subdir), just to keep things clean in case the commands inside the cram test don't properly clean up their temporary files.

Sounds good.